File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        if not nums: return nums
        if len(nums) == 1: return [nums]
        res = []
        for i, c in enumerate(nums):
            x = self.permute(nums[:i] + nums[i+1:])
            for j in x:
                res.append([c, *j])
        return res

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    maxSlidingWindow(nums, k) {
        if (k < 2) return [...nums]
    k = Math.min(k, nums.length)
    const result = []

    for (let n=0; n < nums.length - k + 1; n++){

        let max = -Infinity
        for (let kk=0; kk < k; kk++) {
            max = Math.max(max, nums[n+kk])
        }
        result.push(max)
    }
    return result
    }
}
